add a changelog and update the document for ORC (#20415)
nim-lang/Nim: changelog.md

- Static linking against OpenSSL versions below 1.1, previously done by
  setting `-d:openssl10`, is no longer supported.

> - ORC becomes the default memory management strategy (used to be refc gc). Use
>   `--mm:refc` for a transition period.
> 

Reviewer: ```suggestion
- ORC is now the default memory management strategy. Use
  `--mm:refc` for a transition period.

```
